File: agent/verticals/finance_broker.py
"""Finance broker vertical definition."""

from agent.verticals.base import VerticalDefinition, tabs

VERTICAL = VerticalDefinition(
    key="finance_broker",
    label="Finance Broker",
    risk_level="high",
    entity_label_singular="financial product",
    entity_label_plural="financial products",
    default_plan_label="Finance broker plan",
    crm_tabs=tabs(
        ("overview", "Overview"),
        ("readiness", "Readiness"),
        ("catalog", "Products"),
        ("calculators", "Calculators"),
        ("leads", "Leads"),
        ("compliance", "Compliance"),
        ("activity", "Conversations"),
        ("prompt", "Prompt"),
        ("controls", "Controls"),
    ),
    entity_types=("financial_product", "loan_product", "rate_table", "calculator", "disclosure", "advisor"),
    readiness_checks=("products", "rates", "calculators", "disclosures", "application_flow", "lead_capture"),
    action_types=(
        "SHOW_ENTITIES",
        "COMPARE_ENTITIES",
        "SORT_ENTITIES",
        "RUN_CALCULATOR",
        "RUN_AFFORDABILITY_CALCULATOR",
        "START_APPLICATION",
        "CAPTURE_LEAD",
        "HANDOFF_TO_ADVISOR",
    ),
)

File: crm/src/verticals/definitions/financeBroker.ts
import { tab } from '../shared';
import type { CrmVerticalDefinition } from '../types';

export const financeBrokerVertical: CrmVerticalDefinition = {
  key: 'finance_broker',
  label: 'Finance Broker',
  riskLevel: 'high',
  entityLabelSingular: 'financial product',
  entityLabelPlural: 'financial products',
  defaultPlanLabel: 'Finance broker plan',
  clientTabs: [
    tab('overview', 'Overview'),
    tab('readiness', 'Readiness'),
    tab('catalog', 'Products'),
    tab('calculators', 'Calculators'),
    tab('leads', 'Leads'),
    tab('compliance', 'Compliance'),
    tab('activity', 'Conversations'),
    tab('prompt', 'Prompt'),
    tab('controls', 'Controls'),
  ],
  entityTypes: ['financial_product', 'loan_product', 'rate_table', 'calculator', 'disclosure', 'advisor'],
  readinessChecks: ['products', 'rates', 'calculators', 'disclosures', 'application_flow', 'lead_capture'],
};
